### DISCUSSION TITLE
Compiling without setting Compiler Warnings to Default

### DISCUSSION THREAD

**Discussion by technoblogy** [pschatzmann/arduino-audio-tools#733] [Q&A] [answered]
If you have **Compiler warnings** set to **All** in the Arduino IDE **Preferences** dialogue box, many of the Audio Tools examples will fail to compile. As mentioned in the Audio Tools FAQ, the solution is to set **Compiler warnings** to **Default**. This is because with the **All** option the compile script is set to treat compiler warnings as errors.

If, like me, you prefer not to have to remember to change the preferences, you can edit the file **platform.txt** in the ESP32 installation, which is located at:

    /Users/david/Library/Arduino15/packages/esp32/hardware/esp32/2.0.7/platform.txt

Locate the lines:

    compiler.warning_flags.more=-Wall -Werror=all
    compiler.warning_flags.all=-Wall -Werror=all -Wextra

and change them to:

    compiler.warning_flags.more=-Wall
    compiler.warning_flags.all=-Wall -Wextra

Then you will see the warnings, but they won't cause compilation to fail.

**Comment by pschatzmann** [comment, marked as answer]
Actually you can just change the default and select "More" to get the same result!
Or better correct the warning and submit a pull request.

After my last cleanup there should not be a lot cases left...

**Comment by technoblogy** [comment]
Thanks!

**Reply by pschatzmann** [reply]
I just double checked and corrected the one or two cases.
